src/main/java/org/postgresql/util/PGobject.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Taken from Postgres JDBC Driver 9.4 (BSD license terms)
+ * 
+ * License : https://jdbc.postgresql.org/about/license.html
```

Comment:
I can add copyright same as XAconnection, though I thought it be better to reference the actual license terms for the code, since the short copyright statement doesn't give a clue as to what permissions it's been added under. But maybe that's better left for the LICENSE file

However, changing the namespace raises a bigger issue (se comment 3 above).  
In essence, that would not work for postgis-jdbc, since it has a hard dependency on org.postgresql.util.PGobject. 
It can get this from the postgres jdbc driver, but that would be a bit silly - requiring the driver I'm  trying to swap out still be there runtime, or having the user implement those objects if they care, which is bad usability for this library.

A better solution would then be to go back to having this patch as a separate project, probably under a different namespace, and instead opt for going straight to JTS as an optional dependency in pgjdbc-ng. Can then start weaning people off postgis-jdbc, since it cannot work without these parts of the postgres driver.  JTS is pretty much the default low-level java GIS/topology library which everyone uses (note how hibernate spatial converts postgis-jdbc polygons to JTS polygons) .

Let me know what's your preference